docs/mcp/client.md

```diff
@@ -166,6 +166,44 @@ async def main():
 
 1. See [MCP Run Python](run-python.md) for more information.
 
+## Tool call customisation
+
+The MCP servers provide the ability to set a `process_tool_call` which allows
+the customisation of tool call requests and their responses.
+
+A common use cas for this is to inject metadata to the requests which the server
+call needs.
+
+```python {title="mcp_process_tool_call.py" py="3.10"}
+from typing import Any
+
+from pydantic_ai import Agent
+from pydantic_ai.mcp import CallToolFn, MCPServerStdio, ToolResult
+from pydantic_ai.models.test import TestModel
+from pydantic_ai.tools import RunContext
+
+
+async def process_tool_call(
+    ctx: RunContext[int],
+    tool_call: CallToolFn,
+    tool_name: str,
+    args: dict[str, Any],
+) -> ToolResult:
+    """A noop process_tool_call that just sets a flag."""
```

Comment:
Not a noop since it passes the entire deps to the server!

Reply:
Evolved, will update.

Reply:
```suggestion
    """A tool call processor that passes along the deps."""
```